templates/steamengine_drupal.sh.j2

```diff
@@ -0,0 +1,99 @@
+#!/bin/bash
+
+# {{ ansible_managed }}
+
+display_usage() {
+    echo -e "Usage: $0 <arguments>"
+    echo -e "\nArguments:"
+    echo -e "  start\n\t\t\tStart the http server"
+    echo -e "  stop\n\t\t\tStop the http server"
+    echo -e "  restart\n\t\t\tRestart the http server"
+    echo -e "  status\n\t\t\tDisplay the http server status"
+    echo -e "  reload\n\t\t\tReload the http server configuration"
+    echo -e "  fixpermissions\n\t\tFix permissions of project's files"
+    echo -e "  -h, --help\n\t\t\tShow Usages"
+    echo ""
+}
+
+if [ "$EUID" -ne 0 ]
+then
+    exec sudo "$0" "$@"
+fi
+
+# if less than two arguments supplied, display usage
+if [  $# -le 0 ]
+then
+    display_usage
+    exit 1
+fi
+
+# check whether user had supplied -h or --help . If yes display usage
+if [[ ( $# == "--help") ||  $# == "-h" ]]
+then
+    display_usage
+    exit 0
+fi
+
+while [ $# -gt 0 ]; do
+  arg=$1
+  shift
+  case $arg in
+    start)
+     systemctl start {{ (ansible_os_family == 'Debian') | ternary('php7.4-fpm', 'php-fpm') }}
+     systemctl start nginx
+     exit $?
+    ;;
+    stop)
+     systemctl stop {{ (ansible_os_family == 'Debian') | ternary('php7.4-fpm', 'php-fpm') }}
+     systemctl stop nginx
+     exit $?
+    ;;
+    restart)
+     systemctl restart {{ (ansible_os_family == 'Debian') | ternary('php7.4-fpm', 'php-fpm') }}
+     systemctl restart nginx
+     exit $?
+    ;;
+    status)
+     systemctl status nginx
+     exit $?
+    ;;
+    reload)
+     systemctl reload nginx
+     exit $?
+    ;;
+    fixpermissions)
```

Comment:
There is something wrong with steamengine_project_root_path_web (www directory). Every time I use the role a project, its owner:group is root:root. fixpermissions when executed maually does solve this but deploying a project doest not.

Reply:
Does not fix the permissions of the source of steamengine_drupal_persistent_directory. Could be helpful when restoring Drupal data.

Reply:
La notification est bien faite sur un téléchargement du build. C'est pas ça qui plante de ton côté ?

Reply:
Concerning "steamengine_project_root_path_web (www directory)" the permission issue is adressed by this comment : https://github.com/STEAMULO/steamulo.steamengine/pull/13#discussion_r655406032. Creating the www directory with the right permission solve the issue (not the best way though, as I said all this part must be redone, usage of "command" is a bad pratice. You can clean the www directory and extract the zip at the right place.).

steamengine_drupal_persistent_directory is simply not adressed by the script, as I said it could be helpful to add it.

Reply:
le problème de cette partie, c'est que si on a un problème au moment de la configuration ou autre, le site est HS, pck on a delete l'ancienne version. En préparant la nouvelle avec toute la config, et en basculant au dernier moment, on évite les risques et on réduit le downtime